packages/firebase/externs/firebase-auth-externs.js

```diff
@@ -1043,7 +1043,59 @@ firebase.auth.Auth.prototype.useDeviceLanguage = function() {};
 
 /**
  * Creates a new user account associated with the specified email address and
- * password.
+ * password and returns any additional user info data or credentials.
+ *
+ * On successful creation of the user account, this user will also be
+ * signed in to your application.
+ *
+ * User account creation can fail if the account already exists or the password
+ * is invalid.
+ *
+ * Note: The email address acts as a unique identifier for the user and
+ * enables an email-based password reset.  This function will create
+ * a new user account and set the initial user password.
+ *
+ * <h4>Error Codes</h4>
+ * <dl>
+ * <dt>auth/email-already-in-use</dt>
+ * <dd>Thrown if there already exists an account with the given email
+ *     address.</dd>
+ * <dt>auth/invalid-email</dt>
+ * <dd>Thrown if the email address is not valid.</dd>
+ * <dt>auth/operation-not-allowed</dt>
+ * <dd>Thrown if email/password accounts are not enabled. Enable email/password
+ *     accounts in the Firebase Console, under the Auth tab.</dd>
+ * <dt>auth/weak-password</dt>
+ * <dd>Thrown if the password is not strong enough.</dd>
+ * </dl>
+ *
+ * @example
+ * firebase.auth().createUserAndRetrieveDataWithEmailAndPassword(email, password)
+ *     .catch(function(error) {
+ *       // Handle Errors here.
+ *       var errorCode = error.code;
+ *       var errorMessage = error.message;
+ *       if (errorCode == 'auth/weak-password') {
+ *         alert('The password is too weak.');
+ *       } else {
+ *         alert(errorMessage);
+ *       }
+ *       console.log(error);
+ *     });
+ *
+ * @param {string} email The user's email address.
+ * @param {string} password The user's chosen password.
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.createUserAndRetrieveDataWithEmailAndPassword = function(
+  email,
+  password
+) {};
+
+/**
+ * Creates a new user account associated with the specified email address and
+ * password. Will be depercated, replaced by
```

Comment:
This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#createUserAndRetrieveDataWithEmailAndPassword}.